src/interceptors/XMLHttpRequest/utils/bufferFrom.ts

```diff
@@ -0,0 +1,12 @@
+export function bufferFrom(init: string): Uint8Array {
+  const escstr = encodeURIComponent(init)
+  const binstr = escstr.replace(/%([0-9A-F]{2})/g, function (_, char) {
+    return String.fromCharCode(('0x' + char) as any)
+  })
+  const buffer = new Uint8Array(binstr.length)
+  Array.prototype.forEach.call(binstr, (char, index) => {
+    buffer[index] = char.charCodeAt(0)
+  })
+
+  return buffer
+}
```

Comment:
This is probably a naive question - but does this need to cover any additional cases or be more defensive? I was researching `Buffer.from` ponyfills, and they all handle things quite a bit differently. For example: is it possible for a response body to be a number such as `13` and have a response header type that is erroneously set to `arraybuffer`? Would that break here? I get that it's most likely not the case and is probably an error on the developer's side, but I was curious as to if that should be a concern or not? It seems like we might want to throw TypeErrors or something somewhere if it's misconfigured, but not 100% sure. 

Example: https://github.com/feross/buffer/blob/master/index.js#L166

Reply:
This is a good point. The response body is always retrieved as a string since it's always set as a string (even if you pass a number). I don't think that would cause a problem when transforming `"13"` into a buffer. 

This is a rather dark area, and I think our users will tell us if this implementation works as expected. Note that it's also not specific to this change: we converted the response body to `Buffer` previously, here I'm replacing the native `Buffer` class with this utility for in-browser usage. As always with NRI, I expect regressions and we should cover them with tests.